controller/customdomain_controller.go

```diff
@@ -84,6 +84,22 @@ func (r *CustomDomainReconciler) Reconcile(ctx context.Context, request ctrl.Req
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	clusterVersion, err := r.GetClusterVersion(r.Client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Checking if cluster is using new managed ingress feature")
+	usingNewManagedIngress, err := isUsingNewManagedIngressFeature(r.Client, reqLogger)
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if IsVersionGreaterOrEqualThan(clusterVersion, "4.13") && usingNewManagedIngress {
+		reqLogger.Info("Cluster using new managed ingress feature and is greater than 4.13")
+		return r.returnIngressToClusterIngressOperator(reqLogger, instance)
+	}
```

Comment:
Should we move this new logic below the deletion cleanup logic (specifically, to [here](https://github.com/openshift/custom-domains-operator/pull/120/files#diff-99e5e18e1f4d3aa051dbb732029eafdb783cbf734ea08f305b99aa73afacb6a5L109))?

Reply:
Good point - it does cover the edge case where a Cu decides to delete a CD instead of 'migrate' it to being a 2nd IC. This means that they have the choice of deleting it or letting it become an 'unmanaged' ingress. Will update it now.